widgetssdk/src/main/java/com/glia/widgets/view/unifiedui/extensions/Merge.kt

```diff
@@ -28,15 +28,15 @@ import kotlin.reflect.full.primaryConstructor
  */
 internal inline infix fun <reified T : Any?> T.deepMerge(other: T): T {
     if (!T::class.isData) throw UnsupportedOperationException("Merge supports only data classes")
-    return merge(other)
+    return unsafeMerge(other)
 }
 
 // Added to make method above compatible with Java
 internal fun deepMerge(old: UiTheme?, other: UiTheme?): UiTheme? {
     return old deepMerge other
 }
 
-private fun <T : Any?> T.merge(other: T): T {
+internal fun <T : Any?> T.unsafeMerge(other: T): T {
```

Comment:
Why unsafeMerge? the name suggests that this should not be used?

Reply:
```kotlin
internal inline infix fun <reified T : Any?> T.deepMerge(other: T): T {
    if (!T::class.isData) throw UnsupportedOperationException("Merge supports only data classes")
    return unsafeMerge(other)
}
``` 
deepMerge will throw an exception if it is called on a `non-data class` type(we're using the primary constructor with reflection and the data classes had a primary constructor with at least one argument) and it will prevent unwanted behavior so why it is `safe`, but it is impossible to use inside generic classes/functions if it is not `inline`, so we had to use `unsafe` one, which will not check for data classes.